micro.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+int speekstream16(int i2cfd, int i2caddr, uint16_t addr, uint16_t *data, int size);
+int spokestream16(int i2cfd, int i2caddr, uint16_t addr, uint16_t *data, int size);
+int micro_init(int i2cbus, int i2caddr);
+void spoke16(int i2cfd, int i2caddr, uint16_t addr, uint16_t data);
+uint16_t speek16(int i2cfd, int i2caddr, uint16_t addr);
+int v0_stream_write(int i2cfd, int i2caddr, uint8_t *data, int bytes);
+int v0_stream_read(int i2cfd, int i2caddr, uint8_t *data, int bytes);
+
+typedef enum update_method {
+	UPDATE_V0,
+	UPDATE_V1,
+} update_meth_t;
+
+typedef struct board {
+	const char *compatible;
+	uint16_t modelnum;
+	int i2c_bus;
+	int i2c_chip;
+	update_meth_t method;
+} board_t;
+
+/*
+ * The updates themselves are encrypted/signed, but the below key is just used to
+ * prevent unintentional writes to i2c causing writes to the flash.
+ */
+static const uint32_t magic_key = 0xf092c858;
```

Comment:
This doesn't need to be in a public header. In v0 its buried in code, and in v1 this is used. But no other compilation units really need this. Similar to the state #defines, this could probably be put in a header that is only used by update-v*.c.

Reply:
Fixed, added a update-shared.{c,h}